cwms-data-api/src/main/java/cwms/cda/data/dao/StreamLocationDao.java

```diff
@@ -88,9 +90,14 @@ public List<StreamLocation> retrieveStreamLocations(String streamIdMask, String
      */
     public StreamLocation retrieveStreamLocation(String officeId, String streamId, String locationId, String stationUnit, String stageUnit, String areaUnit) {
         return connectionResult(dsl, conn -> {
-            RETRIEVE_STREAM_LOCATION retrieveStreamLocation = CWMS_STREAM_PACKAGE.call_RETRIEVE_STREAM_LOCATION(DSL.using(conn).configuration(),
-                    locationId, streamId, stationUnit, stageUnit, areaUnit, officeId);
-            return fromJooqStreamLocation(retrieveStreamLocation, locationId, streamId, officeId, stationUnit, stageUnit, areaUnit);
+            try {
+                setOffice(conn, officeId);
+                RETRIEVE_STREAM_LOCATION retrieveStreamLocation = CWMS_STREAM_PACKAGE.call_RETRIEVE_STREAM_LOCATION(DSL.using(conn).configuration(),
+                        locationId, streamId, stationUnit, stageUnit, areaUnit, officeId);
+                return fromJooqStreamLocation(retrieveStreamLocation, locationId, streamId, officeId, stationUnit, stageUnit, areaUnit);
+            } catch (DataAccessException e) {
```

Comment:
use the isNotFoundException helper method in the base class

Reply:
updated to use wrapException helper method, and added missing-stream-location error code to sql exception check.

Reply:
connectionResult should already be doing the wrapException method